cocos/serialization/deserialize.ts

```diff
@@ -1018,54 +1029,47 @@ export function isCompiledJson (json: unknown): boolean {
  * @param options Deserialization Options.
  * @return The original object.
  */
-export function deserialize (data: IFileData | string | CCON | any, details: Details | any, options?: IOptions | any): unknown {
+export function deserialize (data: IDeserializeInput | string | any, details?: Details, options?: IOptions & DeserializeDynamicOptions): unknown {
     if (typeof data === 'string') {
         data = JSON.parse(data);
     }
 
-    const borrowDetails = !details;
-    details = details || Details.pool.get();
+    let isBorrowedDetails = false;
+    if (!details) {
+        const borrowedDetails = Details.pool.get();
+        assertIsTrue(borrowedDetails, `Can not allocate deserialization details`);
+        details = borrowedDetails;
+        isBorrowedDetails = true;
+    }
+
     let res;
 
     if (!FORCE_COMPILED && !isCompiledJson(data)) {
         res = deserializeDynamic(data, details, options);
     } else {
-        details.init(data);
-        options = options || {};
-
-        let version = data[File.Version];
-        let preprocessed = false;
-        if (typeof version === 'object') {
-            preprocessed = version.preprocessed;
-            version = version.version;
-        }
-        if (version < SUPPORT_MIN_FORMAT_VERSION) {
-            throw new Error(getError(5304, version));
-        }
-        options._version = version;
-        options.result = details;
-        data[File.Context] = options;
+        initializeDeserializationContext(
+            data,
+            details,
+            options,
+        );
 
-        if (!preprocessed) {
-            lookupClasses(data, false, options.classFinder, options.reportMissingClass ?? deserialize.reportMissingClass);
-            cacheMasks(data);
-        }
+        const injectedData = data as IIntrudedFileData;
```

Comment:
ditto, runtimeData

Reply:
Renamed as you suggested: https://github.com/cocos/cocos-engine/pull/15892/commits/5d9c25e09854d45a996364f16cdc1a7d7bbd097a